src/components/ContactsList/Merging.vue

```diff
@@ -0,0 +1,648 @@
+		},
+
+		getPropertyValue(property) {
+			if (property.isMultiValue) {
+				// differences between values types :x;x;x;x;x and x,x,x,x,x
+				return property.isStructuredValue
+					? property.getValues()[0]
+					: property.getValues()
+			}
+			return property.getFirstValue()
+		},
+
+		sortUsedProperties() {
+			// the properties where this.conflictInformation[property].type === 'conflict' should have priority
+			return this.usedProperties.sort((a, b) => {
+				const aType = this.conflictInformation[a]?.type
+				const bType = this.conflictInformation[b]?.type
+
+				if (aType === 'conflict' && bType !== 'conflict') {
+					return -1
+				}
+				if (bType === 'conflict' && aType !== 'conflict') {
+					return 1
+				}
+				return 0
+			})
+		},
+
+		async mergeContacts() {
+			const contactToSave = this.contactsList[0]
+
+			const finalProperties = {}
+
+			this.usedProperties.forEach(property => {
+				if (this.conflictInformation[property]?.type === 'conflict') {
+					const resolvedVersion = this.resolvedConflicts.get(property)
+					if (resolvedVersion !== undefined) {
+						 finalProperties[property] = this.dividedProperties[resolvedVersion][property]
+					}
+				} else if (this.conflictInformation[property]?.type === 'conflictWithMultipleValues') {
+					const resolvedVersions = this.resolvedConflicts.get(property)
+					if (resolvedVersions?.size) {
+						finalProperties[property] = Array.from(resolvedVersions).map(version => this.dividedProperties[version][property])
+					}
+				} else if (this.conflictInformation[property]?.type === 'onlyInSecond') {
+					finalProperties[property] = this.dividedProperties[1][property]
+				} else {
+					finalProperties[property] = this.dividedProperties[0][property]
+				}
+			})
+
+			contactToSave.groups = this.selectedGroups
+			contactToSave.addressbook = this.contactsList[this.chosenAddressBook.id].addressbook
+
+			const jCal = contactToSave.jCal
+
+			jCal[1].forEach((property) => {
+				if (Object.keys(finalProperties).includes(property[0]) && finalProperties[property[0]]) {
+					if (Array.isArray(finalProperties[property[0]])) {
+						if (finalProperties[property[0]].length === 1) {
+							property[3] = this.getPropertyValue(finalProperties[property[0]][0])
+						} else {
+							property[3] = this.getPropertyValue(finalProperties[property[0]][0])
+
+							for (let i = 1; i < finalProperties[property[0]].length; i++) {
+								const newProperty = property.slice()
+								newProperty[3] = this.getPropertyValue(finalProperties[property[0]][i])
+								jCal[1].push(newProperty)
+							}
+						}
+					} else {
+						property[3] = this.getPropertyValue(finalProperties[property[0]])
+					}
+				}
+			})
+
+			contactToSave.jCal = jCal
+
+			await this.$store.dispatch('deleteContact', { contact: this.contactsList[0] })
+			await this.$store.dispatch('deleteContact', { contact: this.contactsList[1] })
+
+			await this.$store.dispatch('addContact', contactToSave)
```

Comment:
I can't understand how to modify and then update an existing contact object. Modifying the jCal like this directly is the only thing that I was able to figure out, but is there a better way of doing this?

Also when I go to add the contact I get an error saying `invalid REV`, is it maybe because the URL has to be updated somehow? 

Any help would be appreciated.